helusers/_oidc_auth_impl.py

```diff
@@ -93,6 +93,22 @@ def authenticate_header(self, request):
     def get_audiences(self, api_token):
         return {self.settings.AUDIENCE}
 
+    @staticmethod
+    def _convert_amr_to_list(id_token):
+        """
+        OIDC's amr validation fails, since Tunnistamo sends the amr as a string
+        instead of a list:
+        https://github.com/City-of-Helsinki/tunnistamo/commit/a1b434bbbff92466a144f914a98985008d0ea836.
+        To fix the claim validation issue, convert the id_token amr to list
+        when a string is given.
+        """
+        if isinstance(id_token.get("amr"), str):
+            id_token["amr"] = [id_token["amr"]]
+
+    def validate_claims(self, id_token):
+        self._convert_amr_to_list(id_token)
```

Comment:
How about putting the fix into the `authenticate` method, right after it has successfully decoded the JWT?

Reply:
I'll take a look at it 🧐

Reply:
Yeah, that would make sense. Can just drop the method while I'm at it since it's just two lines. Should be less obfuscating and more readable that way.